platform-tests/src/test/java/org/junit/platform/suite/commons/SuiteLauncherDiscoveryRequestBuilderTests.java

```diff
@@ -214,6 +217,22 @@ class Suite {
 		assertEquals(TestCase.class, exactlyOne(selectors).getJavaClass());
 	}
 
+	@Test
+	void selectMethods() {
+		class TestClass {
+			public void testMethod() {
+			}
+		}
+		@SelectMethod(methodClass = TestClass.class, methodName = "testMethod")
+		class Suite {
+		}
+		LauncherDiscoveryRequest request = builder.suite(Suite.class).build();
+		List<MethodSelector> selectors = request.getSelectorsByType(MethodSelector.class);
+		exactlyOne(selectors);
+		assertTrue(selectors.stream().anyMatch(selector -> selector.getMethodName().equals("testMethod")
+				&& selector.getClassName().equals("TestClass") && selector.getMethodParameterTypes().isEmpty()));
+	}
+
```

Comment:
I removed all the tests to start writing new tests but this test fails and I don't really understand why.

Reply:
@after debugging I notice that if I create and select two methods this test passes, the test fails only if there is just one '@SelectMethod' annotation.

Reply:
After changing the assert_function to 
`assertTrue(selectors.stream().anyMatch(selector -> selector.getMethodName().equals("testMethod")
				&& selector.getClassName().equals(TestClass.class.getName()) && selector.getMethodParameterTypes().equals("")));` 
the test now only fails if there is just one annotation of `@SelectMethod`

Reply:
After debugging for almost an hour,  I finally know what is going on here, the code below just checks if there are multiple annotations of `@SelectMethod`, if there is only one it returns an `emptyList.`

`findAnnotationValues(suiteClass, SelectMethods.class, SelectMethods::value)
				.map(this::selectMethods)
				.ifPresent(this::selectors);`

To solve this problem, I added this code to the suite constructor: 

`		findAnnotation(suiteClass, SelectMethod.class)
				.map(this::selectMethods)
				.ifPresent(this::selectors);`

Is this the best solution?

Reply:
I think using `findRepeatableAnnotations` would be better. The documentation on `findAnnotation` and `findRepeatableAnnotations`  should explain why. 

As an exercise you could also continue with `findAnnotation`. When using `findAnnotation` the next test case you implement, one with multiple `@SelectMethod` annotations will break.

Reply:
As an aside, the `MethodSelector` implements `hasCode()` and `equals()`. So you can use `assertEquals`. This will also give you a nicer error message than using `anyMatch`.

For example:
```
assertEquals(DiscoverySelectors.selectMethod(TestClass.class, "testMethod"), exactlyOne(selectors));

assertEquals(DiscoverySelectors.selectMethod(TestClass.class, "firstMethod"), selectors.get(0));
assertEquals(DiscoverySelectors.selectMethod(TestClass.class, "secondMethod"), selectors.get(1));
```

Reply:
The test case I implemented, one with multiple @SelectMethod annotations passed tough, this is because I used 
```
findAnnotationValues(suiteClass, SelectMethods.class, SelectMethods::value)
				.map(this::selectClasses)
				.ifPresent(this::selectors);
```
In combination with 
`findAnnotation(suiteClass, SelectMethod.class) .map(this::selectMethods) .ifPresent(this::selectors);`

But now I just have this and it works!
		```
findRepeatableAnnotations(suiteClass,SelectMethod.class)
				.stream()
				.map(this::selectMethods)
				.forEach(this::selectors);
```
PS: `findRepeatableAnnotations` does not return an `optional`, instead it returns an empty list if there are no annotations found.